### PR TITLE
Supporting floats

### PR DESCRIPTION
**Pull request by yihozhang**
- [x] introduce the float type at egglog and rust level
- [x] Move `tests/failing/brils/floats` to `passing`.

### CODE REVIEW

File: tests/failing/brils/float/float_args.bril [hunk 1]
```diff
@@ -1,4 +0,0 @@
-# ARGS: 3 3.0 -0.3
```
Comment (yihozhang) [inline tests/failing/brils/float/float_args.bril:1]: This `print` takes several arguments. Should we support it?

Comment (oflatt) [reply tests/failing/brils/float/float_args.bril:1]: Seems okay to skip

File: runtime/rt.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
+#include <stdio.h>
```
Comment (oflatt) [inline runtime/rt.c:1]: Why is this file in c now? We can't use runtime/src/main.rs?

**Review by oflatt — CHANGES_REQUESTED** [review]

File: runtime/rt.c (hunk 2 above)
Comment (yihozhang) [reply runtime/rt.c:1]: We need both c and rust. The bril repo has the `.rs` runtime for brillvm and `.c` runtime for brilift. See `src/util.rs`.

Comment (oflatt) [reply runtime/rt.c:1]: Oh, I see. Thank you

**Pull request merged**